label_studio/tasks/models.py

```diff
@@ -468,10 +466,8 @@ def resolve_uri(self, task_data, project):
 
                 # project storage
                 # TODO: to resolve nested lists and dicts we should improve get_storage_by_url(),
-                # TODO: problem with current approach: it can be used only the first storage that get_storage_by_url
-                # TODO: returns. However, maybe the second storage will resolve uris properly.
-                # TODO: resolve_uri() already supports them
-                storage = self.storage or get_storage_by_url(task_data[field], storage_objects)
+                # Now always using get_storage_by_url to ensure the storage with the correct bucket is used
+                storage = get_storage_by_url(task_data[field], storage_objects)
```

Comment:
What if we use this line? Is it safer, or does it not make sense? As I understand, self.storage exists when a task is linked with a storage using StorageLink. So it makes sense to have the self.storage check as the last resort.. does not it? or I am wrong.. 
```
get_storage_by_url(task_data[field], storage_objects) or self.storage
```